### PR TITLE
added backuper

### PR DESCRIPTION
**Pull request by k1rk**

### CODE REVIEW

File: api/v1beta1/vmcluster_types.go [hunk 1]
```diff
@@ -448,13 +448,80 @@ type VMStorage struct {
...
+	// +optional
+	// +kubebuilder:validation:Enum=INFO;WARN;ERROR;FATAL;PANIC
+	LogLevel *string `json:"logLevel,omitempty"`
+	// Resources container resource request and limits, https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// if not defined default resources from operator config will be used
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Resources"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
+	// +optional
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
```
Comment (f41gh7) [inline api/v1beta1/vmcluster_types.go:505]: lets remove it from open api validation.

File: api/v1beta1/vmcluster_types.go [hunk 2]
```diff
@@ -448,13 +448,80 @@ type VMStorage struct {
...
+	//extra args like maxBytesPerSecond default 0
+	// +optional
+	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
+
+	// Volumes allows configuration of additional volumes on the output deploy definition.
+	// Volumes specified will be appended to other volumes that are generated as a result of
+	// StorageSpec objects.
+	// +optional
+	Volumes []v1.Volume `json:"volumes,omitempty"`
```
Comment (f41gh7) [inline api/v1beta1/vmcluster_types.go]: this one as well

File: api/v1beta1/vmcluster_types.go [hunk 3]
```diff
@@ -448,13 +448,80 @@ type VMStorage struct {
...
+	// VolumeMounts allows configuration of additional VolumeMounts on the output Deployment definition.
+	// VolumeMounts specified will be appended to other VolumeMounts in the container,
+	// that are generated as a result of StorageSpec objects.
+	// +optional
+	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
```
Comment (f41gh7) [inline api/v1beta1/vmcluster_types.go]: and this one.

Comment (k1rk) [reply api/v1beta1/vmcluster_types.go]: you mean to remove ?
```
// +optional
```

File: controllers/factory/vmsingle.go [hunk 4]
```diff
@@ -213,7 +214,6 @@ func makeSpecForVMSingle(cr *victoriametricsv1beta1.VMSingle, c *conf.BaseOperat
 			},
 		})
 	} else {
-
 		volumes = append(volumes, corev1.Volume{
 			Name: "data",
```
Comment (f41gh7) [inline controllers/factory/vmsingle.go]: can you move "data" to const or add  getter method for it  - `.GetBackupVolumeName()`.

**Pull request merged**